zerver/lib/markdown/__init__.py

```diff
@@ -1137,7 +1137,7 @@ def find_proper_insertion_index(
             if (
                 uncle.tag != "div"
                 or "class" not in uncle.keys()
-                or uncle.attrib["class"] not in inline_image_classes
+                or not any(x in uncle.attrib["class"] for x in inline_image_classes)
```

Comment:
Do we need to split this on spaces, to prevent improper substring matches?  Unlikely, but would prevent:
```
uncle = ET.fromstring('<div class="not-an-message_inline_image">baz</div>')
any(x in uncle.attrib["class"] for x in inline_image_classes)
# => True
```

Reply:
Yeah, unlikely to happen, but updated this to be:

`or not any(x in uncle.attrib["class"].split() for x in inline_image_classes)`